wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -103,13 +104,18 @@ public WorkspaceManager(
   public WorkspaceImpl createWorkspace(
       WorkspaceConfig config, String namespace, @Nullable Map<String, String> attributes)
       throws ServerException, NotFoundException, ConflictException, ValidationException {
+    TracingTags.USER_ID.set(() -> EnvironmentContext.getCurrent().getSubject().getUserId());
```

Comment:
Can't we do that in the EnvironementInitializationFilter to avoid adding this top all the sites?

Reply:
+0. Should we have TracingInitializationFilter for that? wdyt @sleshchenko @metlos ?

Reply:
Well, it has two sides:
On one hand: It would mean that we will push `USER_ID` tag to all http requests even when it does not contains any traced method.
On another hand: each traced method would contain user id tags if it is invoked via users request. But it would not have any user id if an operation is done by cron job like activity idling or consistency checker.

I'm OK to leave it as is at least for time being.

Reply:
>I'm OK to leave it as is at least for time being.

+1

Reply:
We already trace all HTTP requests, regardless of whether there is a further more detailed tracing through https://github.com/eclipse/che/blob/master/core/che-core-tracing-web/src/main/java/org/eclipse/che/core/tracing/web/TracingWebModule.java so I see no harm done in adding the user info.

Reply:
So in another words, I agree with @garagatyi that this should be moved to `EnvironmentInitializationFilter`.

Reply:
Ok, I did it in appropriate filters now